cmd/kpod/ps.go

```diff
@@ -0,0 +1,525 @@
+	}
+	server, err := libkpod.New(config)
+	if err != nil {
+		return errors.Wrapf(err, "error creating server")
+	}
+	if err := server.Update(); err != nil {
+		return errors.Wrapf(err, "error updating list of containers")
+	}
+
+	if len(c.Args()) > 0 {
+		return errors.Errorf("too many arguments, ps takes no arguments")
+	}
+
+	format := genPsFormat(c.Bool("quiet"), c.Bool("size"))
+	if c.IsSet("format") {
+		format = c.String("format")
+	}
+	opts := psOptions{
+		all:     c.Bool("all"),
+		filter:  c.String("filter"),
+		format:  format,
+		last:    c.Int("last"),
+		latest:  c.Bool("latest"),
+		noTrunc: c.Bool("no-trunc"),
+		quiet:   c.Bool("quiet"),
+		size:    c.Bool("size"),
+	}
+
+	containers, err := server.ListContainers()
+	if err != nil {
+		return errors.Wrapf(err, "error getting containers from server")
+	}
+	var params *FilterParamsPS
+	if opts.filter != "" {
+		params, err = parseFilter(opts.filter, containers)
+		if err != nil {
+			return errors.Wrapf(err, "error parsing filter")
+		}
+	} else {
+		params = nil
+	}
+
+	containerList := getContainersMatchingFilter(containers, params, server)
+
+	return psOutput(containerList, server, opts)
+}
+
+// generate the template based on conditions given
+func genPsFormat(quiet, size bool) (format string) {
+	if quiet {
+		return formats.IDString
+	}
+	format = "table {{.ID}}\t{{.Image}}\t{{.Command}}\t{{.CreatedAt}}\t{{.Status}}\t{{.Ports}}\t{{.Names}}\t"
+	if size {
+		format += "{{.Size}}\t"
+	}
+	return
+}
+
+func psToGeneric(params []psOutputParams) []interface{} {
+	genericParams := make([]interface{}, len(params))
+	for i, v := range params {
+		genericParams[i] = interface{}(v)
+	}
+	return genericParams
+}
+
+// generate the accurate header based on template given
+func (p *psOutputParams) headerMap() map[string]string {
+	v := reflect.Indirect(reflect.ValueOf(p))
+	values := make(map[string]string)
+
+	for i := 0; i < v.NumField(); i++ {
+		key := v.Type().Field(i).Name
+		value := key
+		if value == "ID" {
+			value = "Container" + value
+		}
+		values[key] = strings.ToUpper(splitCamelCase(value))
+	}
+	return values
+}
+
+func psOutput(containers []*libkpod.ContainerData, server *libkpod.ContainerServer, opts psOptions) error {
+	var (
+		output           []psOutputParams
+		containersOutput []*libkpod.ContainerData
+		status           string
+		ctrID            string
+		command          string
+		runningFor       string
+		imageName        string
+		mounts           string
+		ports            string
+		size             string
+		labels           string
+		createdAt        string
+	)
+
+	if len(containers) == 0 {
+		return nil
+	}
+	if opts.last > 0 {
+		if opts.last > len(containers) {
+			containersOutput = containers
+		} else {
+			for i := 0; i < opts.last; i++ {
+				containersOutput = append(containersOutput, containers[i])
+			}
+		}
+	} else if opts.latest {
+		containersOutput = append(containersOutput, containers[0])
+	} else if !opts.all {
```

Comment:
I think this is a bug, they should be mutually exclusive.  
--latest means print latest container (option should not exist, since it is easy to do with head/tail.
Same with --last.
But lets make them mutually exclusive and not rely on some hidden ovverride.